eis_toolkit/exploratory_analyses/k_means_cluster.py

```diff
@@ -0,0 +1,68 @@
+from typing import Optional
+
+import geopandas as gdp
+import numpy as np
+from beartype import beartype
+from sklearn.cluster import KMeans
+
+from eis_toolkit.exceptions import EmptyDataFrameException, InvalidParameterValueException
+
+
+def _k_means_clustering(
+    data: gdp.GeoDataFrame, number_of_clusters: Optional[int], random_state: Optional[int]
+) -> gdp.GeoDataFrame:
+
+    coordinates = list(data.geometry.apply(lambda geom: [geom.x, geom.y]))
+
+    if number_of_clusters is None:
+        # The elbow method
+        k_max = 10
+        inertia = np.array(
+            [KMeans(n_clusters=k, random_state=0).fit(coordinates).inertia_ for k in range(1, k_max + 1)]
+        )
+
+        inertia = np.diff(inertia, 2)
+        scaled_derivatives = [i * 100 for i in inertia]
+        k_optimal = scaled_derivatives.index(min(scaled_derivatives))
+
+        kmeans = KMeans(n_clusters=k_optimal, random_state=random_state)
+
+    else:
+        kmeans = KMeans(n_clusters=number_of_clusters, random_state=random_state)
+
+    kmeans.fit(coordinates)
+    data["cluster"] = kmeans.labels_
+
+    return data
+
+
+@beartype
+def k_means_clustering(
+    data: gdp.GeoDataFrame, number_of_clusters: Optional[int] = None, random_state: Optional[int] = None
+) -> gdp.GeoDataFrame:
+    """
+    Perform k-means clustering on the input data.
+
+    Args:
+        gdf: A GeoDataFrame containing the input data.
+        number_of_clusters: The number of clusters to form. Optional parameter.
```

Comment:
Add here a short mention about the elbow-method when no value for number_of_clusters is provided and also a mentioning about valid values (at least 1)